src/main/java/in/projecteka/consentmanager/user/OtpAttemptRepository.java

```diff
@@ -0,0 +1,56 @@
+package in.projecteka.consentmanager.user;
+
+import in.projecteka.consentmanager.common.DbOperationError;
+import in.projecteka.consentmanager.user.model.OtpAttempt;
+import io.vertx.pgclient.PgPool;
+import io.vertx.sqlclient.Tuple;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Repository;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Repository
+@AllArgsConstructor
+public class OtpAttemptRepository {
+
+    private static final String INSERT_OTP_ATTEMPT = "INSERT INTO " +
+            "otp_attempt (phone_number,blocked) VALUES ($1,$2)";
```

Comment:
you need to have when was the last failure attempt, if it was done long ago you should reset the count, and for what action (phonenumber verification/account linking,etc) also matters.

Reply:
So we are checking the timestamp of last blocked attempt that is similar to having when was the last failure.

Reply:
`For what action` needs to be done.